main.go

```diff
@@ -74,6 +84,21 @@ func getEnvAsBool(name string, defaultValue bool) bool {
 }
 
 func main() {
+
+	// ----- fetch controller namespace -----
+	controllerNamespace := os.Getenv(ControllerNamespaceEnvVar)
+	if controllerNamespace == "" {
+		bytes, err := os.ReadFile(KubeNamespaceFile)
+		if err != nil {
+			//TODO check kube context and retrieve namespace from there
+			setupLog.Info("Error reading Kube-mounted namespace file, reverting to default namespace",
+				"file", KubeNamespaceFile, "err", err, "default", DefaultControllerNamespace)
+			controllerNamespace = DefaultControllerNamespace
```

Comment:
I think this should be a `panic`. Although the default is fair, it is probably better to make no assumptions.